zebedee-cms/src/main/java/com/github/onsdigital/zebedee/keyring/LegacyKeyringImpl.java

```diff
@@ -6,104 +6,224 @@
 import com.github.onsdigital.zebedee.session.model.Session;
 import com.github.onsdigital.zebedee.session.service.Sessions;
 import com.github.onsdigital.zebedee.user.model.User;
+import com.github.onsdigital.zebedee.user.service.UsersService;
 import org.apache.commons.lang3.StringUtils;
 
 import javax.crypto.SecretKey;
 import java.io.IOException;
+import java.util.HashSet;
 import java.util.Set;
 
+import static com.github.onsdigital.zebedee.logging.CMSLogEvent.info;
+
 /**
- * This class duplicates the existing (legacy) keyring functionality behind a newly defined Keyring interface.
- * Moving the keyring functionality behind an interface allows to swap implemetations more easily.
+ * This class duplicates the existing (legacy) {@link com.github.onsdigital.zebedee.json.Keyring} and {@link KeyringCache}
+ * functionality behind a newly defined Keyring interface. Doing so allows to start the process of migrating to
+ * the new {@link Keyring} abstraction while maintaing backwards compatability and the ability to swicth back if
+ * necessary.
  */
 public class LegacyKeyringImpl implements Keyring {
 
     static final String USER_NULL_ERR = "user required but was null";
     static final String EMAIL_EMPTY_ERR = "user email required but was empty";
     static final String USER_KEYRING_NULL_ERR = "user keyring expected was null";
     static final String GET_SESSION_ERR = "error getting user session";
-    static final String SESSION_NULL_ERR = "user session required but was null";
     static final String CACHE_PUT_ERR = "error added user keys to legacy keyring cache";
     static final String COLLECTION_NULL_ERR = "collection required but was null";
     static final String COLLECTION_DESC_NULL_ERR = "collection description required but was null";
     static final String COLLECTION_ID_EMPTY_ERR = "collection ID required but was null/empty";
     static final String SECRET_KEY_NULL_ERR = "secret key required but was null";
-    static final String USER_KEYRING_LOCKED_ERR = "unable to access locked user keyring";
+    static final String CACHE_GET_ERR = "error getting entry from keyringCache";
+    static final String SAVE_USER_ERR = "error persiting user update";
```

Comment:
Consider defining more specific types for the different error messages. This would make it cleaner to catch specific error types at a higher level. The tests could then check for the specific error type instead of a specific message. (may be overkill)